NDTensors/src/blocksparse/blocksparse.jl

```diff
@@ -51,16 +41,41 @@ end
 function BlockSparse(
   ::Type{ElT}, ::UndefInitializer, blockoffsets::BlockOffsets, dim::Integer; vargs...
 ) where {ElT<:Number}
-  return BlockSparse(Vector{ElT}(undef, dim), blockoffsets; vargs...)
+  return BlockSparse(default_datatype(ElT)(undef, dim), blockoffsets; vargs...)
+end
+
+function BlockSparse(::UndefInitializer, blockoffsets::BlockOffsets, dim::Integer; vargs...)
+  return BlockSparse(NDTensors.default_eltype(), undef, blockoffsets, dim; vargs...)
 end
 
 function BlockSparse(blockoffsets::BlockOffsets, dim::Integer; vargs...)
-  return BlockSparse(Float64, blockoffsets, dim; vargs...)
+  elt = default_eltype()
+  datat = default_datatype(elt)
+  N = ndims(datat)
+  return BlockSparse(UnallocatedZeros{elt,N,datat}, blockoffsets, dim; vargs...)
```

Comment:
Probably should not use `UnallocatedZeros` here.

Reply:
Okay, I replaced this with `default_datatype(default_eltype())`